src/traces/scattergl/convert.js

```diff
@@ -360,10 +360,11 @@ function convertLinePositions(gd, trace, positions) {
     };
 }
 
-function convertErrorBarPositions(gd, trace, positions) {
+function convertErrorBarPositions(gd, trace, pos, x, y) {
     var calcFromTrace = Registry.getComponentMethod('errorbars', 'calcFromTrace');
-    var vals = calcFromTrace(trace, gd._fullLayout);
-    var count = positions.length / 2;
+    var _trace = Lib.extendFlat({}, trace, {x: x, y: y});
```

Comment:
This seems weird to me - putting arrays of `calcdata` values into a fake trace. I don't know if category or date data would make any sense in this context but if we *were* to support them this would break.

This is the only caller of `calcFromTrace`, right? Can we just pass in the x and y calc'ed arrays separately from the trace?

Reply:
Ah nice - that's more of a refactor than I was expecting but you're right, much better. 🌴 before, but as well as the 🐎 overhead, the scaffolding meant it didn't even really result in shorter code.